src/utilities/constants.ts

```diff
@@ -60,7 +60,17 @@ export const CONTRACT_XVS_TOKEN_ADDRESS = isOnTestnet
   : contracts.MAIN.TOKEN.xvs;
 
 // Contract ABI
-export const CONTRACT_TOKEN_ADDRESS = isOnTestnet
+export const CONTRACT_TOKEN_ADDRESS: Record<
+  string,
+  {
+    id: string;
+    symbol: string;
+    address: string | undefined;
+    decimals: number;
+    asset: string;
+    vasset?: string;
+  }
```

Comment:
Actually, this type is partially incorrect: the key of the record isn't a `string`, but a set of possible string values. This is important because we'll some accuracy by typing it as a `string` only. Perhaps we can extract all the token names into a `string` and use this value as key instead?

```typescript
type TokenSymbol = 'sxp' | 'xvs' | 'bnb'; // TODO: add all token symbols

type Token = {
  id: TokenSymbol,
  symbol: Uppercase<TokenSymbol>,
  decimals: number,
  address: string,
  asset: string,
  vasset?: string,
};

export const CONTRACT_TOKEN_ADDRESS: Record<TokenSymbol, Token> = {...};
```